chris_backend/feeds/views.py

```diff
@@ -355,9 +355,12 @@ def update_owners(self, serializer):
         """
         Custom method to update the feed's owners.
         """
+        
         feed = self.get_object()
         owners = feed.owner.values('username')
         username = self.request.data.pop('owner')
+        if not isinstance(self.request.data, dict) or not getattr(self.request.data, '_mutable', 1):
```

Comment:
Line 362 must come before line 361 (and probable before line 360 as well), the point of the `if` statement is to check whether it's possible to call `self.request.data.pop`, and terminate early if not possible.

Continuing on the discussion from here: https://github.com/FNNDSC/ChRIS_ultron_backEnd/pull/388/files#r863826517

I looked into it and found a better solution. Albeit more inefficient, I think it's best that we work within the confines of the public API rather than causing trouble down the line by accessing private variables.

https://stackoverflow.com/questions/44717442/this-querydict-instance-is-immutable/51196932#51196932

Is mutability strictly necessary here anyways? Why do we need to pop, maybe just getting the value would work?

Seeing that this same code is copy-pasted in multiple places, please put it in a helper function, or think about how we can use object-oriented inheritance to solve this problem with less code duplication. As a challenge, can you think of how we can use better design so that this kind of error wouldn't be a concern altogether?

Important: please add unit test to cover the bug.